templates/CMakeLists.txt.template

```diff
@@ -322,6 +331,19 @@
   include(cmake/upb.cmake)
   include(cmake/xxhash.cmake)
   include(cmake/zlib.cmake)
+  include(cmake/xds_proto.cmake)
+
+  % for external_library in external_libraries:
+  # Setup external proto libraray at ${external_library.destination}
+  if (NOT EXISTS <%text>${CMAKE_CURRENT_SOURCE_DIR}</%text>/${external_library.destination})
```

Comment:
+1 for checking the submodule is already there and skipping the download if so. Perhaps we should add a comment that explains why exactly you're doing this?